### ISSUE TITLE
service module doesn't seem to restart, it seems to stop/start

### ISSUE THREAD

**Issue by benchub** [ansible/ansible-modules-core#1836]
It seems when I tell the service module to restart a process, it doesn't actually tell it to restart, it tells it to stop/start. I understand that many services don't see a difference, but they are not the same! For example, with the pgbouncer connection pooler, "restart" means, "restart the process, but first make sure we copy all the socket information so that we don't have to make clients reconnect." If you stop/start instead, you force all your clients to reconnect.

I know I can just use a command to do "service pgbouncer restart" instead, but it would be nice if that's what service's state=restarted did. At the very least, if this is a "won't fix" issue, then at least change the docs to make it clear that "restarted" is actually done with a stop/start.

**Comment by dirtyharrycallahan** [comment]
start/stop follows the convention of applications conforming to LSB.

As defined in LSB 4.0 Core generic

restart stop and restart the service if the service is already running, otherwise start the service

Ref: http://refspecs.linuxbase.org/LSB_4.0.0/LSB-Core-generic/LSB-Core-generic/iniscrptact.html

**Comment by jlebon** [comment]
This also causes an issue for systemd. When we call `systemctl restart foo`, systemd will first stop services that have `foo` in their `Requires=` field, and after restarting `foo`, will also restart those dependent services. Right now, those dependent services do not get restarted.

Couldn't we just use `restart` as long as no sleep is given? SysV scripts may not have restart, but systemd (and surely upstart) should be able to handle it fine.

**Comment by jameshartig** [comment]
We're having the same issue as @jlebon which means we can never use `service: state=restarted` since it breaks random services that `Requires` that service. `systemd` automatically handles stopping/starting when you call `restart` so why is ansible doing that itself?

I'll make a PR to fix it if that's really the only thing holding this up?

**Comment by ansibot** [comment]
@grossws, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@grossws, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/30076
